Assets/Scripts/Draggable.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(Rigidbody))]
+[RequireComponent(typeof(NonConvexMeshCollider))]
+public class Draggable : MonoBehaviour
+{
+    static int InteractLayer = 10;
+
+    private void OnValidate()
+    {
+        transform.tag = "Draggable";
+        gameObject.layer = InteractLayer;
+
+        //MeshCollider meshCollider = GetComponent<MeshCollider>();
+        //if (meshCollider != null)
+        //{
+        //    DestroyImmediate(meshCollider);
+        //}
+
+        Rigidbody rb = GetComponent<Rigidbody>();
+        if (rb == null)
+        {
+            rb = gameObject.AddComponent<Rigidbody>();
+        }
```

Comment:
this isn't needed if you do requirescomponent. Requirescomponent will force auto add them to an object on assignment if it doesn't have it. Same for the other one below.